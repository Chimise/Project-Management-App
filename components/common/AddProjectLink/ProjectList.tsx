import React from "react";
import {
  DocumentIcon,
  EllipsisHorizontalIcon,
} from "@heroicons/react/24/solid";
import Link from "next/link";
import type { ProjectSchema } from "../../../models/Project";

interface ProjectListProps {
  project: ProjectSchema;
  onClick: (event: React.MouseEvent<any>) => void;
}

const ProjectList = ({ project, onClick }: ProjectListProps) => {
    
  return (
    <span onClick={onClick} className="flex w-full items-center pr-6 pl-5 text-sm font-medium py-2 text-gray-600 hover:bg-black hover:text-white">
      <div className="flex-1 inline-flex items-center flex-nowrap space-x-1">
        <DocumentIcon className="w-4 h-4" />
        <span className="truncate">{project.name}</span>
      </div>
      <EllipsisHorizontalIcon className="w-4 h-4" />
    </span>
  );
};

export default ProjectList;
