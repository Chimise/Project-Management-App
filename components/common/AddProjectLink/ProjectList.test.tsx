import '@testing-library/jest-dom';
import {render, user} from '../../../test/test-utils';
import ProjectList from "./ProjectList";
import router, {mockUseRouter, mockPush} from '../../../test/router';

beforeEach(() => {
  jest.useFakeTimers();
  mockPush.mockClear();
  
});
afterEach(() => {
  jest.useRealTimers();
});

mockUseRouter();


const project = {
    id: 1,
    tasks: [],
    user_id: 1,
    created_at: new Date().toDateString(),
    updated_at: new Date().toISOString(),
    name: "New Project"
}

test('Checks that the project list components remains unchanged', () => {
    const {container} = render(<ProjectList project={project} onClick={jest.fn()} />);
    expect(container.firstChild).toMatchSnapshot();
})

test('Checks that the projects render properly and clicking the project redirects the user', async () => {
    const handler = jest.fn(() => mockPush('/projects/1'));
    const {container} = render(<ProjectList project={project} onClick={handler} />)
    await user.click(container.firstElementChild!);
    expect(handler).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalled();
    expect(mockPush).toHaveBeenCalledWith('/projects/1');
});

test('The project name is rendered appropriately', () => {
  const {getByText} = render(<ProjectList project={project} onClick={jest.fn()} />)
  const projectName = getByText('New Project');
  expect(projectName).toBeInTheDocument();
})