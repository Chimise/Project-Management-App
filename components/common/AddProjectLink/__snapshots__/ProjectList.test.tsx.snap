// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the project list components remains unchanged 1`] = `
<span
  class="flex w-full items-center pr-6 pl-5 text-sm font-medium py-2 text-gray-600 hover:bg-black hover:text-white"
>
  <div
    class="flex-1 inline-flex items-center flex-nowrap space-x-1"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.625 1.5c-1.036 0-1.875.84-1.875 1.875v17.25c0 1.035.84 1.875 1.875 1.875h12.75c1.035 0 1.875-.84 1.875-1.875V12.75A3.75 3.75 0 0016.5 9h-1.875a1.875 1.875 0 01-1.875-1.875V5.25A3.75 3.75 0 009 1.5H5.625z"
      />
      <path
        d="M12.971 1.816A5.23 5.23 0 0114.25 5.25v1.875c0 .207.168.375.375.375H16.5a5.23 5.23 0 013.434 1.279 9.768 9.768 0 00-6.963-6.963z"
      />
    </svg>
    <span
      class="truncate"
    >
      New Project
    </span>
  </div>
  <svg
    aria-hidden="true"
    class="w-4 h-4"
    fill="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M4.5 12a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zm6 0a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zm6 0a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0z"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;
